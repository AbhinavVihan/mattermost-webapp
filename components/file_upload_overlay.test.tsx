// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import FileUploadOverlay from 'components/file_upload_overlay';
import {renderWithIntl} from 'tests/react_testing_utils';

describe('components/FileUploadOverlay', () => {
    test('should match snapshot when file upload is showing with no overlay type', () => {
        const wrapper = renderWithIntl(
            <FileUploadOverlay
                overlayType=''
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when file upload is showing with overlay type of right', () => {
        const wrapper = renderWithIntl(
            <FileUploadOverlay
                overlayType='right'
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot when file upload is showing with overlay type of center', () => {
        const wrapper = renderWithIntl(
            <FileUploadOverlay
                overlayType='center'
            />,
        );

        expect(wrapper).toMatchSnapshot();
    });
});
