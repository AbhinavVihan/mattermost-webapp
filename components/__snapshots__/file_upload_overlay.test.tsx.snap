// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with no overlay type 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-overlay hidden"
      >
        <div
          class="overlay__indent"
        >
          <div
            class="overlay__circle"
          >
            <img
              alt="Files"
              class="overlay__files"
              src="fileOverlayImage"
            />
            <span>
              <i
                class="fa fa-upload"
                title="Upload Icon"
              />
              Drop a file to upload it.
            </span>
            <img
              alt="Logo"
              class="overlay__logo"
              src="overlayLogoImage"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-overlay hidden"
    >
      <div
        class="overlay__indent"
      >
        <div
          class="overlay__circle"
        >
          <img
            alt="Files"
            class="overlay__files"
            src="fileOverlayImage"
          />
          <span>
            <i
              class="fa fa-upload"
              title="Upload Icon"
            />
            Drop a file to upload it.
          </span>
          <img
            alt="Logo"
            class="overlay__logo"
            src="overlayLogoImage"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with overlay type of center 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-overlay hidden center-file-overlay"
      >
        <div
          class="overlay__indent"
        >
          <div
            class="overlay__circle"
          >
            <img
              alt="Files"
              class="overlay__files"
              src="fileOverlayImage"
            />
            <span>
              <i
                class="fa fa-upload"
                title="Upload Icon"
              />
              Drop a file to upload it.
            </span>
            <img
              alt="Logo"
              class="overlay__logo"
              src="overlayLogoImage"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-overlay hidden center-file-overlay"
    >
      <div
        class="overlay__indent"
      >
        <div
          class="overlay__circle"
        >
          <img
            alt="Files"
            class="overlay__files"
            src="fileOverlayImage"
          />
          <span>
            <i
              class="fa fa-upload"
              title="Upload Icon"
            />
            Drop a file to upload it.
          </span>
          <img
            alt="Logo"
            class="overlay__logo"
            src="overlayLogoImage"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`components/FileUploadOverlay should match snapshot when file upload is showing with overlay type of right 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="file-overlay hidden right-file-overlay"
      >
        <div
          class="overlay__indent"
        >
          <div
            class="overlay__circle"
          >
            <img
              alt="Files"
              class="overlay__files"
              src="fileOverlayImage"
            />
            <span>
              <i
                class="fa fa-upload"
                title="Upload Icon"
              />
              Drop a file to upload it.
            </span>
            <img
              alt="Logo"
              class="overlay__logo"
              src="overlayLogoImage"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="file-overlay hidden right-file-overlay"
    >
      <div
        class="overlay__indent"
      >
        <div
          class="overlay__circle"
        >
          <img
            alt="Files"
            class="overlay__files"
            src="fileOverlayImage"
          />
          <span>
            <i
              class="fa fa-upload"
              title="Upload Icon"
            />
            Drop a file to upload it.
          </span>
          <img
            alt="Logo"
            class="overlay__logo"
            src="overlayLogoImage"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
